metricbeat/module/elasticsearch/index/data.go

```diff
@@ -32,17 +32,25 @@ var (
 	}
 )
 
-func eventsMapping(r mb.ReporterV2, info elasticsearch.Info, content []byte) {
+func eventsMapping(r mb.ReporterV2, info elasticsearch.Info, content []byte) []error {
```

Comment:
This looks a bit confusing to me, do we need two ways of reporting errors? (One with the reporter api `r.Error`, and another one returning the list of errors).

Reply:
It's two different types of errors. The ones with `r.Error` will be send as error events to Elasticsearch. The `[]error` here is only used for testing. I agree, it's not optimal and we should come up with a cleaner approach in the future.

Reply:
Ok